pkg/raft/raft.go

```diff
@@ -1066,12 +1066,18 @@ func (r *raft) reset(term uint64) {
 
 	r.abortLeaderTransfer()
 
-	r.electionTracker.ResetVotes()
 	r.trk.Visit(func(id pb.PeerID, pr *tracker.Progress) {
+		guessMatch := r.raftLog.lastIndex()
+		hintEntry, ok := r.electionTracker.GetHintEntry(id)
+		conflictIndex, t := r.raftLog.findConflictByTerm(hintEntry.Index, hintEntry.Term)
+		if ok && t != 0 {
+			guessMatch = conflictIndex
+		}
```

Comment:
This part shouldn't be in every `raft.reset`. We only need to do it when `becomeLeader`.